src/main/com/smithsmodding/armory/common/logic/initialization/CommonSystemInitializer.java

```diff
@@ -181,6 +123,38 @@ private static void tryRemoveRecipeFromGame(@Nonnull IRecipe recipe, @Nonnull It
         }
     }
 
+    @Override
+    public void onPreInit(@Nonnull final FMLPreInitializationEvent preInitializationEvent)
+    {
+        registerCreativeTabs();
+    }
+
+    private static void registerCreativeTabs()
+    {
+        ModCreativeTabs.GENERAL = new GeneralTabs();
+        ModCreativeTabs.COMPONENTS = new ComponentsTab();
+        ModCreativeTabs.HEATEDITEM = new HeatedItemTab();
+        ModCreativeTabs.ARMOR = new ArmorTab();
+    }
+
+    @Override
+    public void onInit(@Nonnull final FMLInitializationEvent initializationEvent)
+    {
+        registerTileEntities();
+        setupCreativeTabs();
+    }
+
+    private void registerTileEntities()
+    {
+        GameRegistry.registerTileEntity(TileEntityForge.class, new ResourceLocation(References.General.MOD_ID, References.InternalNames.TileEntities.ForgeContainer));
+        GameRegistry.registerTileEntity(TileEntityFireplace.class, new ResourceLocation(References.General.MOD_ID, References.InternalNames.TileEntities.FireplaceContainer));
+        GameRegistry.registerTileEntity(TileEntityBlackSmithsAnvil.class, new ResourceLocation(References.General.MOD_ID, References.InternalNames.TileEntities.ArmorsAnvil));
+        GameRegistry.registerTileEntity(TileEntityConduit.class, new ResourceLocation(References.General.MOD_ID, References.InternalNames.TileEntities.Conduit));
+        GameRegistry.registerTileEntity(TileEntityMoltenMetalTank.class, new ResourceLocation(References.General.MOD_ID, References.InternalNames.TileEntities.Tank));
+        GameRegistry.registerTileEntity(TileEntityPump.class, new ResourceLocation(References.General.MOD_ID, References.InternalNames.TileEntities.Pump));
+        GameRegistry.registerTileEntity(TileEntityMoltenMetalMixer.class, new ResourceLocation(References.General.MOD_ID, References.InternalNames.TileEntities.MoltenMetalMixer));
+    }
+
     private static void initializeOreDict() {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Move this left curly brace to the beginning of next line of code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3ALeftCurlyBraceStartLineCheck)